graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/SchemaProcessor.java

```diff
@@ -210,22 +211,24 @@ private GraphQL buildGraphql(
     // the queries we generate, they're not useful for users of the schema.
     schema = removeCqlDirectives(schema);
 
-    com.apollographql.federation.graphqljava.SchemaTransformer federationTransformer =
-        Federation.transform(schema);
+    GraphQL.Builder graphqlBuilder;
     if (mappingModel.hasFederatedEntities()) {
-      federationTransformer
-          .fetchEntities(new FederatedEntityFetcher(mappingModel))
-          .resolveEntityType(
-              environment -> {
-                FederatedEntity entity = environment.getObject();
-                return environment.getSchema().getObjectType(entity.getTypeName());
-              });
+      GraphQLSchema federationReadySchema =
+          Federation.transform(schema)
+              .fetchEntities(new FederatedEntityFetcher(mappingModel))
+              .resolveEntityType(
+                  environment -> {
+                    FederatedEntity entity = environment.getObject();
+                    return environment.getSchema().getObjectType(entity.getTypeName());
+                  })
+              .build();
+      graphqlBuilder =
+          GraphQL.newGraphQL(federationReadySchema)
+              .instrumentation(new FederatedTracingInstrumentation());
```

Comment:
isn't the `instrumentation` costly? If so, shouldn't we set it only if some setting is enabled? (and allow to disable that if needed)

Reply:
It is already conditioned by an HTTP header (see the integration test).

Reply:
So is this `apollo-federation-include-trace` handled internally in the `FederatedTracingInstrumentation`?

Reply:
Correct, the HTTP header is handled internally [here](https://github.com/apollographql/federation-jvm/blob/master/graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/tracing/FederatedTracingInstrumentation.java#L469-L476).